src/main/java/org/killbill/billing/plugin/adyen/client/payment/builder/checkout/CheckoutPaymentsBuilder.java

```diff
@@ -151,16 +155,28 @@ private void setShopperData() {
         if (userData.getGender() != null) {
             name.setGender(Name.GenderEnum.valueOf(userData.getGender().toUpperCase()));
         }
+
         if (userData.getFirstName() != null ||
             userData.getInfix() != null ||
             userData.getLastName() != null ||
             userData.getGender() != null) {
             request.setShopperName(name);
         }
 
-        request.setShopperLocale(userData.getShopperLocale().toString());
+        // Split the locale text with only 2 chars
+        // e.g. Use 'de' if given locale is de_DE
+        if(userData.getShopperLocale() != null) {
+            final String localeText = userData.getShopperLocale().toString();
+            String locale = localeText;
+            if(localeText.contains("_")) {
+                String[] localeParts = localeText.split("_");
+                request.setShopperLocale(localeParts.length > 0 ? localeParts[0] : localeText);
+            } else {
+                request.setShopperLocale(localeText);
+            }
+        }
+
         request.setShopperEmail(userData.getShopperEmail());
```

Comment:
Any reason to remove the shopperIP?

Reply:
Not required for Klarna/Checkout API. Why to send more data than needed?

Reply:
Did we asked this from Adyen. What impact Adyen or Banks can have with missing Customer IP ?
I think we are passing as part of Adyen requests already, why to remove it from Checkout API.

I think we should ask it from `Adyen` for the impact of it.